inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.cpp

```diff
@@ -165,57 +162,86 @@ void MKLDNNIfNode::initSupportedPrimitiveDescriptors() {
     supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::unknown);
 }
 
-
 void MKLDNNIfNode::createPrimitive() {
-    const auto& eng = getEngine();
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
+    }
+}
 
-    for (auto& map_rule : thenInputPortMap) {
-        auto &fromMem = getParentEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
-        auto &toMem = inputMemThen[map_rule.to];
+bool MKLDNNIfNode::needPrepareParams() const {
+    return MKLDNNNode::inputShapesDefined() || condition != (reinterpret_cast<const uint8_t*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr()))[0];
+}
 
-        beforeThenMappers.emplace_back(std::make_shared<PortMapHelper>(fromMem, toMem, eng));
+void MKLDNNIfNode::prepareParams() {
+    const auto& eng = getEngine();
+
+    if (isDynamic) {
+        new_state = true;
+        condition = (reinterpret_cast<const uint8_t*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr()))[0];
```

Comment:
we still can store `condition` as `bool`
```suggestion
        condition = static_cast<bool>((reinterpret_cast<const uint8_t*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr()))[0]);
```

Reply:
Ok, done. Thank you